### PR TITLE
Create jira api client bundle and remove bloat

### PR DESCRIPTION
**Pull request by arothuis**
[Ticket #104424786](https://www.pivotaltracker.com/story/show/104424786)

Also remove the boilerplate default app bundle, controllers and routes.

### CODE REVIEW

**Comment by arothuis** [comment]
Would like a review: @DRvanR en @rjkip

File: app/config/config.yml [hunk 1]
```diff
@@ -19,8 +19,6 @@ framework:
     csrf_protection: ~
     validation:      { enable_annotations: true }
     #serializer:      { enable_annotations: true }
-    templating:
-        engines: ['twig']
```
Comment (DRvanR) [inline app/config/config.yml]: Does this [still work][1]?

[1]: http://symfony.com/doc/current/reference/configuration/framework.html#engines

File: src/Surfnet/JiraApiClientBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+     * {@inheritdoc}
+     */
+    public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder();
+        // $rootNode = $treeBuilder->root('jira_api_client');
+
+        // Here you should define the parameters that are allowed to
+        // configure your bundle. See the documentation linked above for
+        // more information on that topic.
```
Comment (DRvanR) [inline src/Surfnet/JiraApiClientBundle/DependencyInjection/Configuration.php]: Please remove obsolete comments :wink:

File: src/Surfnet/JiraApiClientBundle/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace Surfnet\JiraApiClientBundle\DependencyInjection;
+
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+
+/**
+ * This is the class that validates and merges configuration from your app/config files
+ *
+ * To learn more see {@link http://symfony.com/doc/current/cookbook/bundles/extension.html#cookbook-bundles-extension-config-class}
```
Comment (DRvanR) [inline src/Surfnet/JiraApiClientBundle/DependencyInjection/Configuration.php]: this is something that is obsolete as well :wink:

File: src/Surfnet/JiraApiClientBundle/DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ */
+class Configuration implements ConfigurationInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder();
+        // $rootNode = $treeBuilder->root('jira_api_client');
```
Comment (DRvanR) [inline src/Surfnet/JiraApiClientBundle/DependencyInjection/Configuration.php]: AFAIK, you should definitely add the root, as the `TreeBuilder` itself is used. However, the return value is not used, as we add nothing to our tree. Worth a check I'd say...

**Comment by DRvanR** [comment]
Seem to be missing the .gitignore additions?

File: app/config/config.yml (hunk 1 above)
Comment (arothuis) [inline app/config/config.yml]: > At least one engine is required.

Interesting, while we don't really *need* templating, we need a templating engine?

Comment (DRvanR) [inline app/config/config.yml]: we might not need it, but SF2 might :wink: (can imagine they need it for exception screens in dev, the toolbar, etc.)

File: src/Surfnet/JiraApiClientBundle/DependencyInjection/Configuration.php (hunk 4 above)
Comment (arothuis) [inline src/Surfnet/JiraApiClientBundle/DependencyInjection/Configuration.php]: Good point. PHPMD can be a stickler about unused variables (```$rootNode```), which is why I commented it out. 
However, ```$treeBuilder->root()``` sets the root for the ```TreeBuilder```. When the ```buildTree()``` method is called (which happens when configurations are processed), the root property is expected to be ```!null``` or else it wil throw a RuntimeException.

I figure we don't have to store the return value in a variable, so I will remove ```$rootNode```.

Comment (DRvanR) [inline src/Surfnet/JiraApiClientBundle/DependencyInjection/Configuration.php]: hehe, exactly what I was aiming at :wink:

**Comment by arothuis** [comment]
AFAIK, .gitignore does not have to be changed because the files in question would be removed by this PR.

**Comment by DRvanR** [comment]
and it would be created on [`composer install`][1] / [`composer update`][2] (albeit the latter is broken - should be fixed as well :wink)

[1]: https://github.com/SURFnet/conext-operations-support/blob/develop/composer.json#L34
[2]: https://github.com/SURFnet/conext-operations-support/blob/develop/composer.json#L43

**Comment by arothuis** [comment]
Removed comments, unused variables, Composer install/update scripts. Added php as templating engine.

**Comment by DRvanR** [comment]
Not sure if removal of SymfonyRequirements (rather than ignoring) actually hinders us when upgrading... But hey ho, this has already taken too much time. Merge away. :+1:

**Pull request merged**